src/arrayMethodJoin.js

```diff
@@ -5,6 +5,26 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
+    let str = '';
+
+    if (this.length === 0) {
+      return str;
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (separator === undefined) {
+        str += this[i] + ',';
+      } else if (this[i] === null || this[i] === undefined) {
+        this[i] = separator;
+        str += this[i];
+      } else if (separator === null) {
+        str += this[i] + 'null';
+      } else {
+        str += this[i] + separator;
```

Comment:
```suggestion
        str = str + this[i] + separator;
```
і все працюватиме навіть для null, а чому то ти мені розкажеш)

Reply:
Так и не поняла, почему += не работает )